skbeam/core/tests/test_roi.py

```diff
@@ -341,3 +341,27 @@ def test_kymograph():
     # number
     for row in kymograph_data:
         assert np.all(row == row[0])
+
+
+def test_bars_boxes():
+    h_values, v_values = np.mgrid[0:10, 0:10]
+    edges = [[3, 4], [5, 7]]
+
+    h_label_array = roi.bar_rois(edges, h_values)
+    v_label_array = roi.bar_rois(edges, v_values)
+
+    assert_array_equal(h_label_array[3, :], np.ones((10,), dtype=np.int))
+    assert_array_equal(v_label_array[:, 3], np.ones((10,), dtype=np.int))
+    assert_array_equal(np.unique(h_label_array)[1:], np.array([1, 2]))
+    assert_array_equal(np.unique(v_label_array)[1:], np.array([1, 2]))
+
+    box_array = roi.box_rois(v_values, edges, h_values)
+
+    assert_array_equal(box_array[3, 3], 1)
+    assert_array_equal(box_array[5, 5], 4)
+    assert_array_equal(np.unique(box_array)[1:], np.array([1, 2, 3, 4]))
+
+    n_edges = edges = [[3, 4], [5, 7], [8]]
+    h_val, v_val = np.mgrid[0:20, 0:20]
+    assert_raises(ValueError, lambda: roi.box_rois(v_values, n_edges))
+    assert_raises(ValueError, lambda: roi.box_rois(v_values, edges, h_val))
```

Comment:
Drop the `lambda`. `assert_raises` works like this:
```python
assert_raises(Exception, callable, *args, **kwargs)
```
Under the hood assert_raises can be understood to be a context manager that calls `callable(*args, **kwargs)` and checks that `ValueError` was raised, so you should write your `assert_raises` like this:

```python
assert_raise(ValueError, roi.box_rois, v_values, edges, h_val)
```